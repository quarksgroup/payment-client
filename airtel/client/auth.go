package client

import (
	"context"

	"github.com/quarksgroup/payment-client/airtel"
)

//login responsible client api authentication to your airtel account portal but doesn't exposed outside
func (c *Client) login(ctx context.Context, id, secret, grantType string) (*airtel.Token, *airtel.Response, error) {
	endpoint := "auth/oauth2/token"
	in := tokenRequest{
		ClientId:     id,
		ClientSecret: secret,
		GrantType:    grantType,
	}
	out := new(tokenResponse)
	res, err := c.do(ctx, "POST", endpoint, in, out, nil)
	return convertToken(out), res, err
}

type tokenRequest struct {
	ClientId     string `json:"client_id"`
	ClientSecret string `json:"client_secret"`
	GrantType    string `json:"grant_type"`
}

type tokenResponse struct {
	AccessToken string `json:"access_token"`
	TokenType   string `json:"token_type"`
	Expires     int64  `json:"expires_in"`
}

func convertToken(tk *tokenResponse) *airtel.Token {

	// expires := time.Now().Local().Add(180 * time.Second)

	return &airtel.Token{
		Token:   tk.AccessToken,
		Type:    tk.TokenType,
		Expires: tk.Expires,
	}
}
