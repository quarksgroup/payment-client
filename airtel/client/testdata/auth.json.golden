{
    "token_type": "bearer",
    "access_token": "hntg00kF8WnArnobVUl6KG1xlRssUl2k",
    "expires_in": "2022-07-04 11:51:48.59491863 +0200 CAT"
}